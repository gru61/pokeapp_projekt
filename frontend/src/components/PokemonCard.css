.pokemon-card {
    background-color: var(--color-card-bg);
    color: var(--color-input-text);
    border-radius: 10px;
    box-shadow: 0 2px 5px rgba(0,0,0,0.2);
    text-align: center;
    padding: 1rem;
    transition: transform 0.2s, box-shadow 0.2s;
    cursor: pointer;
}

.pokemon-card:hover {
    transform: scale(1.05);
    box-shadow: 0 4px 10px rgba(0,0,0,0.3);
}

.pokemon-image {
    width: 100px;
    height: 100px;
    object-fit: contain;
    margin-bottom: 0.5rem;
}

.pokemon-name {
    font-size: 1rem;
    color: var(--color-input-text);
}