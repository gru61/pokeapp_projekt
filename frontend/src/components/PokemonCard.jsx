import "./PokemonCard.css";
import {useNavigate} from "react-router-dom";

function PokemonCard({ name, id }) {
    const navigate = useNavigate();
    const paddedId = id.toString().padStart(3, "0");
    const spriteUrl = `/sprites/${paddedId}.png`;


    const handleClick = () => {
        navigate(`/species/${id}`);
    };

    return (
        <div className="pokemon-card" onClick={handleClick}>
            <img
                src={spriteUrl}
                alt={name}
                className="pokemon-image"
                onError={(e) => {
                    e.target.src = "/sprites/placeholder.png"; // fallback image
                }}
            />
            <h3 className="pokemon-name">{name}</h3>
        </div>
    );
}

export default PokemonCard;