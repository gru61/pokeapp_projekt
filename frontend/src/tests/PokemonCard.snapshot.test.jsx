import '@testing-library/jest-dom';
import { render } from "@testing-library/react";
import PokemonCard from "../components/PokemonCard.jsx";

const mon = {
    id: 1,
    pokedexId: 25,
    speciesName: "Pikachu",
    nickname: "Pika",
    boxName: "Team",
    level: 7,
    edition: "Rot",
    type1: "ELEKTRO",
};

test("Snapshot von PokemonCard", () => {
    const { asFragment } = render(<PokemonCard mon={mon} />);
    expect(asFragment()).toMatchSnapshot();
});
