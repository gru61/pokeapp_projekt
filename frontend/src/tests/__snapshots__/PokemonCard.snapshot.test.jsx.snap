// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot von PokemonCard 1`] = `
<DocumentFragment>
  <div
    class="pokemon-card edition-rot"
  >
    <div
      class="pokemon-meta-topleft"
    >
      <span
        class="pokemon-edition"
      >
        Rot
      </span>
      <span
        class="pokemon-boxname"
      >
        Team
      </span>
    </div>
    <div
      class="pokemon-speciesid"
    >
      #025
    </div>
    <img
      alt="Pikachu"
      class="pokemon-sprite"
      src="/sprites/025.png"
    />
    <div
      class="pokemon-name"
    >
      <span
        class="pokemon-nickname"
      >
        Pika
      </span>
      <span
        class="pokemon-realname"
      >
        Pikachu
      </span>
    </div>
    <div
      class="pokemon-level-inline"
    >
      Lvl. 7
    </div>
    <div
      class="pokemon-types"
    >
      <span
        class="poke-type elektro"
      >
        ELEKTRO
      </span>
    </div>
  </div>
</DocumentFragment>
`;
